diff --git a/01-basic b/01-basic-remove-first-line
index 30c5fc9cb..4f0b01321 100644
--- a/01-basic
+++ b/01-basic-remove-first-line
@@ -1,5 +1,4 @@
-/*---------------------------------------------------------------------------------------------
- *  Copyright (c) Microsoft Corporation and GitHub. All rights reserved.
+*  Copyright (c) Microsoft Corporation and GitHub. All rights reserved.
  *--------------------------------------------------------------------------------------------*/
 import { spawn } from 'child_process';
 
